[TF FE] Update docs for TF FE (#17453)
openvinotoolkit/openvino: docs/resources/tensorflow_frontend.md

@sphinxdirective

> TensorFlow Frontend is C++ based Frontend for conversion of TensorFlow models. MO now uses the TensorFlow Frontend as the default path for conversion to IR.

Reviewer: ```suggestion
TensorFlow Frontend is a C++ based Frontend for converting TensorFlow models. MO now uses the TensorFlow Frontend as the default path for conversion to IR.
```

> Also, the frontend allows to load TensorFlow models (in SavedModel, MetaGraph, and frozen Protobuf formats) directly via ``read_model()`` method.

Reviewer: ```suggestion
Also, the frontend allows loading TensorFlow models in SavedModel, MetaGraph, and frozen Protobuf formats directly via the ``read_model()`` method.
```

> Also, the frontend allows to load TensorFlow models (in SavedModel, MegatGraph, and frozen Protobuf formats) directly via ``read_model()`` method.

Reviewer 2: ``
```suggestion
Also, the frontend allows to load TensorFlow models (in SavedModel, MetaGraph, and frozen Protobuf formats) directly via ``read_model()`` method.
```

The current limitations:

* IRs generated by new TensorFlow Frontend are compatible only with OpenVINO API 2.0

> * There is no full parity between the legacy frontend and new frontend in MO. Known limitations compared to the legacy approach are:

Reviewer: ```suggestion
* There is no full parity between the legacy frontend and the new frontend in MO. Known limitations compared to the legacy approach are:
```

TF1 Control flow, Complex types, models requiring config files and old python extensions. The solution detects unsupported functionalities and provides fallback.

> To force using the legacy frontend, ``--use_legacy_frontend`` must be specified.

Reviewer: Maybe "to force the use of the legacy fronted"? Force using gives an impression like force something with the help of the frontend

Author: Fixed.

> To force using the legacy frontend, ``--use_legacy_frontend`` must be specified.

Reviewer: ```suggestion
To force the use of the legacy frontend, ``--use_legacy_frontend`` must be specified.
```
